src/components/ChatList/GroupChatChannelItem.tsx

```diff
@@ -21,18 +21,22 @@ const GroupChatChannelItem = (props: ChannelItemProps) => {
   const hasBadge = unreadCount > 0;
 
   const description = groupChat?.rawJson?.firstMessage ?? groupChat?.rawJson?.message;
-  const isSystemDescription = description?.isSystem ?? description?.type === 'system';
+  const isSystemDescription =
+    description?.isSystem ||
+    description?.type === 'system' ||
+    description?.message?.type === 'system';
   const isDeletedMessage = description?.type === 'deleted';
   const sender = description?.user?.username ?? description?.user?.name;
-  const isMeAsSender = description?.user?.id === signedProfileId;
+  const isMeAsSender =
+    description?.user?.id === signedProfileId || description?.user?.isMe || !description;
 
   let channelDescription = groupChat?.description;
-  if (!isSystemDescription) {
+  if (!isSystemDescription && description !== channelDescription) {
     channelDescription = isMeAsSender
       ? `You: ${channelDescription}`
       : `${sender}: ${channelDescription}`;
   }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The handling of `channelDescription` has been refactored. The new code only updates `channelDescription` if `description` is not equal to `channelDescription`, which might prevent unnecessary updates. However, this could potentially introduce bugs if `description` and `channelDescription` are supposed to be different in some cases.
```diff
- if (!isSystemDescription) {
-    channelDescription = isMeAsSender
-      ? `You: ${channelDescription}`
-      : `${sender}: ${channelDescription}`;
-  }
- if (!sender || isDeletedMessage) channelDescription = '';
+ if (!isSystemDescription && description !== channelDescription) {
+   channelDescription = isMeAsSender
+     ? `You: ${channelDescription}`
+     : `${sender}: ${channelDescription}`;
+ }
+ if (isDeletedMessage) channelDescription = '';
```

Please ensure that these changes are intended and thoroughly tested to avoid potential issues.


<!-- This is an auto-generated comment by OSS CodeRabbit -->